arangod/RestHandler/RestStatusHandler.cpp

```diff
@@ -276,3 +287,26 @@ RestStatus RestStatusHandler::executeOverview() {
   generateResult(rest::ResponseCode::OK, result.slice());
   return RestStatus::DONE;
 }
+
+RestStatus RestStatusHandler::executeMemoryProfile() {
+#if defined(USE_MEMORY_PROFILE)
+  bool found;
+  std::string outDirectory = _request->value("directory", found);
+
+  if (!found) {
+    outDirectory = TRI_GetTempPath();
+  }
+
+  std::string const fileName = FileUtils::buildFilename(outDirectory, "profile.out");
+  char const* f = fileName.c_str();
+  mallctl("prof.dump", NULL, NULL, &f, sizeof(const char *));
+  std::string const content = FileUtils::slurp(fileName);
+
+  resetResponse(rest::ResponseCode::OK);
+
+  _response->setContentType(rest::ContentType::TEXT);
+  _response->addRawPayload(velocypack::StringRef(content));
+
+  return RestStatus::DONE;
+#endif
```

Comment:
I suggest to take the `return` statement out of the #if to avoid compiler warnings about nothing being returned.